### DISCUSSION TITLE
Inconsistent errors while using Python scripting module

### DISCUSSION THREAD

**Discussion by vchundru2002** [bonsai-rx/bonsai#1723] [Q&A]
Hi, I've been trying to use the Python scripting functionality of Bonsai, but every few runs (inconsistently, Bonsai throws this error:
![image](https://github.com/bonsai-rx/bonsai/assets/60073624/95c75bc3-462e-4451-b699-2300ef869af7)

Afterwards, Bonsai shuts down. All I do in the python script is import some modules:
![image](https://github.com/bonsai-rx/bonsai/assets/60073624/764c2ff8-1cf4-4e55-bd88-31ae15fe99a9)

My Bonsai workflow is pretty simple too:
![image](https://github.com/bonsai-rx/bonsai/assets/60073624/ec1761e1-b47e-49cd-ba2f-8e33340378f4)

This occurs even when I open the script in a module instead of the runtime, however, when the script is empty, I don't get this error. 
I haven't been able to figure out what the issue is. Any ideas or debugging tips?

**Comment by bruno-f-cruz** [comment]
Append your workflow and Python code files. Thanks

**Reply by vchundru2002** [reply]
[BonsaiPythonScripts.zip](https://github.com/bonsai-rx/bonsai/files/14812747/BonsaiPythonScripts.zip)

I can send my python environment as well. It was too big to upload here.

**Reply by vchundru2002** [reply]
Not sure if this helps but I noticed that if I leave the workflow running for at least 10s before hitting stop, I never get this error. However, if I don't wait long enough, then the error comes. Oddly, the Python script doesn't take 10s to run, it finishes running in under 1s, but if I stop the workflow at about 3s it'll usually throws this error. Is there some kind of issue when setting up and tearing down the Python interpreter in Bonsai? How would I get around this? Is there a safer way to stop the workflow?
